### PR TITLE
Add empty email filter to the Data-flow Consent Service call

### PR DESCRIPTION
**Pull request by LaurenQurashi**
### Description of change

In 2016, A change was made to the model which made emails mandatory. This leaves us with around 40,000 pre 2016 made contacts, with no emails attached to them. This would break the Consent Service when it was passed an empty email from the Data Flow pipeline, which calls on this service periodically. 

This small PR filters out empty string emails and discards them from the call made to the consent service. This seemed like a sensible choice as a contact with no email attached to it, can't accept email marketing. A test has also been added to check that this filtering works as it should.

### Checklist

* [x] If this is a releasable change, has a news fragment been added?

  <details>
  <summary>Explanation</summary>
  
  A news fragment is required for any releasable change (i.e. code that runs in or affects production) so that a corresponding changelog entry is added when releasing.
  
  Check [changelog/README.md](https://github.com/uktrade/data-hub-api/blob/master/changelog/README.md) for instructions.
  
  </details>
  
* [x] Has this branch been rebased on top of the current `develop` branch?

  <details>
  <summary>Explanation</summary>
  
  The branch should not be stale or have conflicts at the time reviews are requested.
  
  </details>

* [x] Is the CircleCI build passing?

### General points

<details>
<summary><strong>Other things to check</strong></summary><p></p>

* Make sure `fixtures/test_data.yaml` is maintained when updating models
* Consider the admin site when making changes to models
* Use select-/prefetch-related field lists in views and search apps, and update them when fields are added
* Make sure the README is updated e.g. when adding new environment variables

</details>

See [docs/CONTRIBUTING.md](https://github.com/uktrade/data-hub-api/blob/develop/docs/CONTRIBUTING.md) for more guidelines.

### CODE REVIEW

File: datahub/dataset/contact/test/test_views.py [hunk 1]
```diff
@@ -127,3 +127,29 @@ def test_makes_api_call_to_consent_service(
...
+    ):
+        """
+        Test that empty email strings are removed before making a call to the Consent Service.
+        Before 2016, emails for contacts were not mandatory, so there are around 40,000
+        records with no emails. Filtering has been added to make sure that the
+        Consent Service is not called with these empty emails.
+        """
+        FeatureFlagFactory(code=GET_CONSENT_FROM_CONSENT_SERVICE, is_active=True)
+        contact1 = ContactFactory(email='a@a.a')
+        contact2 = ContactFactory(email='')
```
Comment (sekharpanja) [inline datahub/dataset/contact/test/test_views.py:144]: Are we sure there are no `None`s along with blanks?

Comment (LaurenQurashi) [reply datahub/dataset/contact/test/test_views.py:144]: Hey! @adamledwards confirmed there are no emails set to none, just empty strings

Comment (adamledwards) [reply datahub/dataset/contact/test/test_views.py:144]: There are no `None`, I can confirm that the DB a non-null constraint on it. So it looks like the migration placeholder where set to an empty string

File: changelog/contact/add-filtering-to-data-flow-consent-service-call.bugfix.md [hunk 2]
```diff
@@ -0,0 +1 @@
+In 2016, a change was made to the models which made emails mandatory for contacts. Prior to this, over 40,000 contacts had been created with no emails attached to them. Data-Flow periodically sends requests to the Consent Service for a list of contacts' marketing preferences. When one of these contact's empty emails gets included in the request, it breaks the Consent Service. To fix this, a filter has been added in the viewset Data-Flow uses, to prevent empty email strings from being sent to the Consent Service.
```
Comment (adamledwards) [inline changelog/contact/add-filtering-to-data-flow-consent-service-call.bugfix.md:1]: 👍

**Pull request merged**